### DISCUSSION TITLE
Best way to use Pin Mode while switching between Laptop and Monitors

### DISCUSSION THREAD

**Discussion by prabhavtiwari** [rxhanson/RectanglePro-Community#266] [Q&A]
Hi All,

What would be the best way to setup pin mode if I use my laptop with and without external monitor.
It just works fine on an external monitor with small space taken by the pinned app, but when I disconnect the monitor on the laptop screen it doesn't work too well with an app pinned, leaving small space for other apps to run.
So I have to manually turn off pin mode and it's not the best experience.

Any way to turn off pin mode when external monitor is disconnected, or other workarounds?

**Comment by rxhanson** [comment]
Thanks for the feedback. What would be your desired experience here? That it automatically turns off pin mode on disconnect, or that it has a size configured per display, or something else?

**Reply by prabhavtiwari** [reply]
I think the pin mode makes more sense on external monitors. I doubt anyone would use that on just mac inbuilt screen, cuz a very few apps can be resized to the smallest size and then everything would be out of proportion if there is a app pinned all the time.

So it would be great to have the pin mode trigger only on external monitor is connected. So just issue a pin mode enable and reflow as soon as mac is connected to a monitor and back to a normal laptop when disconnected. Having this automation would make things a lot easier for everyone who switches their displays.
